### PR TITLE
[Merged by Bors] - feat: `T2Space` is `Antitone`

### PR DESCRIPTION
**Pull request by scholzhannah**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — APPROVED** [review]
Thank you for the PR, this is definitely a missing lemma. Your proof is locally optimal, good job!
Rewriting by a different lemma, a slightly shorter proof is possible. I am personally not sure which one is nicer, and will let the maintainers decide.

File: Mathlib/Topology/Separation.lean [hunk 1]
```diff
@@ -1472,6 +1472,13 @@ theorem Set.InjOn.exists_isOpen_superset {X Y : Type*} [TopologicalSpace X] [Top
   let ⟨u, huo, hsu, hut⟩ := mem_nhdsSet_iff_exists.1 hst
   ⟨u, huo, hsu, ht.mono hut⟩
 
+theorem t2Space_antitone {X : Type*} : Antitone (@T2Space X) := by
+  intro s t le t2
+  rw [t2Space_iff] at t2 ⊢
+  intro i j ne
+  obtain ⟨u, v, openu, openv, huv⟩ := t2 ne
```
[the remark this answers is not in the document]
Comment (j-loreaux) [reply Mathlib/Topology/Separation.lean:1479]: If you move this further down in the file, you can make use of a nice lemma:
```
theorem t2Space_antitone {X : Type*} : Antitone (@T2Space X) :=
  fun inst₁ inst₂ h_top h_t2 ↦ @T2Space.of_injective_continuous _ _ inst₁ inst₂
    h_t2 _ Function.injective_id <| continuous_id_of_le h_top
```

**Comment by j-loreaux** [comment]
@scholzhannah can you explain why you want this lemma anyway? Are you using it in another PR?

It's not a big deal to add it I suppose, but it seems like it wouldn't come in handy very often (if ever).

**Comment by scholzhannah** [comment]
> @scholzhannah can you explain why you want this lemma anyway? Are you using it in another PR?
> 
> It's not a big deal to add it I suppose, but it seems like it wouldn't come in handy very often (if ever).

I use it for the product of CW-complexes (which I hope will be a PR at some point). I need to define a finer topology on the product and have it still be a Hausdorff space. But I think I am only using it that one time and I don't know how useful that lemma is in general.

File: Mathlib/Topology/Separation.lean [hunk 2]
```diff
@@ -1472,6 +1472,13 @@ theorem Set.InjOn.exists_isOpen_superset {X Y : Type*} [TopologicalSpace X] [Top
   let ⟨u, huo, hsu, hut⟩ := mem_nhdsSet_iff_exists.1 hst
   ⟨u, huo, hsu, ht.mono hut⟩
 
+theorem t2Space_antitone {X : Type*} : Antitone (@T2Space X) := by
+  intro s t le t2
```
[the remark this answers is not in the document]
Comment (scholzhannah) [reply Mathlib/Topology/Separation.lean]: Thank you, that's good to know!

**Review by j-loreaux — COMMENTED** [review]
That's fine then, thanks!

bors merge

**Pull request closed without merging**